app/src/main/java/com/github/se/project/ui/lesson/AddLesson.kt

```diff
@@ -1,239 +1,43 @@
-              }
-
-              item {
-                Row(modifier = Modifier.fillMaxWidth()) {
-                  Button(
-                      onClick = { datePickerDialog.show() },
-                      modifier = Modifier.weight(1f).testTag("DateButton")) {
-                        Text(
-                            selectedDate.ifEmpty { "Select Date" },
-                            style = MaterialTheme.typography.labelMedium)
-                      }
-                  Spacer(modifier = Modifier.width(12.dp))
-                  Button(
-                      onClick = { timePickerDialog.show() },
-                      modifier = Modifier.weight(1f).testTag("TimeButton")) {
-                        Text(
-                            selectedTime.ifEmpty { "Select Time" },
-                            style = MaterialTheme.typography.labelMedium)
-                      }
-                }
-              }
-
-              item {
-                Spacer(modifier = Modifier.height(16.dp))
-                Text(
-                    "Select the subject you want to study",
-                    style = MaterialTheme.typography.titleSmall)
-                SubjectSelector(selectedSubject)
-              }
-
-              item {
-                Spacer(modifier = Modifier.height(16.dp))
-                Text(
-                    "Select the possible languages for the course",
-                    style = MaterialTheme.typography.titleSmall)
-                LanguageSelector(selectedLanguages)
-              }
-
-              item {
-                Spacer(modifier = Modifier.height(16.dp))
-                PriceRangeSlider("Select a price range for your lesson:") { min, max ->
-                  minPrice = min.toDouble()
-                  maxPrice = max.toDouble()
-                }
-                Text("Selected price range: ${minPrice.toInt()} - ${maxPrice.toInt()}.-")
-              }
-            }
-      },
-      bottomBar = {
-        Button(
-            onClick = onConfirm,
-            modifier = Modifier.fillMaxWidth().padding(14.dp).testTag("confirmButton")) {
-              Text("Confirm your request", fontSize = 16.sp)
-            }
-      })
-}
-
-fun validateLessonInput(
-    title: String,
-    description: String,
-    selectedSubject: MutableState<Subject>,
-    selectedLanguages: List<Language>,
-    date: String,
-    time: String
-): String? {
-  for (entry in
-      mapOf(
-              "title" to title,
-              "description" to description,
-              "subject" to selectedSubject.value.name,
-              "language" to selectedLanguages.joinToString { it.name },
-              "date" to date,
-              "time" to time,
-          )
-          .entries) {
-    if (entry.value.isEmpty()) {
-      return "${entry.key} is missing"
-    }
-  }
-  return null
+  LessonEditor(
+      mainTitle = "Schedule a new lesson",
+      profile = profile.value!!,
+      lesson = null,
```

Comment:
This is strange in my opinion to pass null to a function especially that in LessonEditor you use lesson!!, the process will probably fail please adjust it